src/libfetchers/tarball.cc

```diff
@@ -365,6 +365,16 @@ struct TarballInputScheme : CurlInputScheme
 
         return {result.accessor, input};
     }
+
+    std::optional<std::string> getFingerprint(ref<Store> store, const Input & input) const override
+    {
+        if (auto narHash = input.getNarHash())
+            return narHash->to_string(HashFormat::SRI, true);
+        else if (auto rev = input.getRev())
+            return rev->gitRev();
```

Comment:
Doesn't it also support `revCount` or `lastModified`?
In the `git` fetcher we may assume that `revCount` is functionally dependent on `rev`, but that's not necessarily the case for an arbitrary http endpoint that may or may not have broken behavior.

Reply:
Good point, I've changed `LockedFlake::getFingerprint()` to add the `revCount` and `lastModified` to the fingerprint.